app/src/main/java/com/eveningoutpost/dexdrip/ui/helpers/BitmapUtil.java

```diff
@@ -86,6 +56,84 @@ public static Bitmap getTiled(final Bitmap source, final int xSize, final int yS
+
+                Bitmap image_bitmap = null;
+                if (background_file.startsWith("content://")) {
+                    FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor();
+                    image_bitmap = BitmapFactory.decodeFileDescriptor(fileDescriptor);
+                } else {
+                    File file = new File(background_file);
+                    FileInputStream in = new FileInputStream(file);
+                    image_bitmap = BitmapFactory.decodeStream(in);
+                    in.close();
+                }
+
+                Bitmap rotated_bitmap = image_bitmap;
+                final Matrix image_matrix = new Matrix();
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.N) {
+                    ExifInterface exif = null;
+                    if (background_file.startsWith("content://")) {
+                        FileDescriptor fileDescriptor = xdrip.getAppContext().getContentResolver().openFileDescriptor(background_uri, "r").getFileDescriptor(); // reset
+                        exif = new ExifInterface(fileDescriptor);
+                    } else {
+                        File file = new File(background_file);
+                        FileInputStream in = new FileInputStream(file);
+                        exif = new ExifInterface(in);
+                        in.close();
+                    }
+                    int rotation = exifOrientationToDegrees(exif.getAttributeInt(ExifInterface.TAG_ORIENTATION, ExifInterface.ORIENTATION_NORMAL));
+                    android.util.Log.d("NumberWall", "Rotation: " + rotation);
+
+                    if (rotation != 0) {
+                        image_matrix.preRotate(rotation);
+                        rotated_bitmap = Bitmap.createBitmap(image_bitmap, 0, 0, image_bitmap.getWidth(), image_bitmap.getHeight(), image_matrix, true);
+                        image_bitmap.recycle();
+                    }
+                }
+
+                tiledBitmap = getBestCroppedScaled(rotated_bitmap, xSize, ySize);
+                BitmapLoader.saveBitmapAsBundle(background_file, Bitmap.createBitmap(tiledBitmap));
+            } else {
+                tiledBitmap = Bitmap.createBitmap(tiledBitmap); // make a copy
+                android.util.Log.d("NumberWall", "cache hit");
+            }
+        } catch (Exception e) {
+            // cannot load bitmap
+            android.util.Log.e("NumberWall", "Cannot load bitmap: " + e);
+        }
+        return tiledBitmap;
+    }
+
+    public static String copyBackgroundImage(Uri sourceUri) {
+        File destinationFolder = new File(xdrip.getAppContext().getFilesDir().getAbsolutePath() + File.separator + "numberWall");
+        destinationFolder.mkdirs();
+        String destinationFileName = "background" + new Date().getTime() + ".png"; // unique name to avoid caching
```

Comment:
What happens with the PNGs created during a previous session? Or more precisely, where and when are the unique background images cleaned?

Reply:
yeah, you are right! My first implementation was just overriding the same `background.png` file, but that was not working because of the bitmap caching. I'll add a call to clean that folder before saving the new file.